src/operators.jl

```diff
@@ -1,121 +1,130 @@
-import Compat: @functorize
+## Lifted operators
 
-@noinline throw_error() = error()
+importall Base.Operators
+import Base: promote_op, abs, abs2, sqrt, cbrt, scalarmin, scalarmax
+using Compat: @functorize
 
-for f in (
-    :(@compat Base.:+),
-    :(@compat Base.:-),
-    :(@compat Base.:!),
-    :(@compat Base.:~),
-)
-    @eval begin
-        @inline function $(f){S}(x::Nullable{S})
-            if isbits(S)
-                Nullable($(f)(x.value), x.isnull)
-            else
-                throw_error()
-            end
-        end
-    end
+# Methods adapted from Base Julia 0.5
+if VERSION <= v"0.5.0-dev"
+    promote_op(::Any, T) = T
+    promote_op{T}(::Type{T}, ::Any) = T
+
+    promote_op{R<:Number}(op, ::Type{R}) = typeof(op(one(R)))
+    promote_op{R<:Number,S<:Number}(op, ::Type{R}, ::Type{S}) = typeof(op(one(R), one(S)))
 end
 
-# Implement the binary operators: +, -, *, /, %, &, |, ^, <<, and >>
-for f in (
-    :(@compat Base.:+),
-    :(@compat Base.:-),
-    :(@compat Base.:*),
-    :(@compat Base.:/),
-    :(@compat Base.:%),
-    :(@compat Base.:&),
-    :(@compat Base.:|),
-    :(@compat Base.:^),
-    :(@compat Base.:<<),
-    :(@compat Base.:>>),
-)
+"""
+    null_safe_op(f::Any, ::Type)::Bool
+    null_safe_op(f::Any, ::Type, ::Type)::Bool
+
+Returns whether an operation `f` can safely be applied to any value of the passed type(s).
```

Comment:
Maybe we should mention "bit pattern" below instead of "domain"? I think "value" is good as a first approximation to give the idea of the requirements.